pkg/datastore/test/counters.go

```diff
@@ -65,6 +67,41 @@ func RelationshipCounterOverExpiredTest(t *testing.T, tester DatastoreTester) {
 	require.Equal(t, expectedCount, count)
 }
 
+func RegisterRelationshipCountersInParallelTest(t *testing.T, tester DatastoreTester) {
+	rawDS, err := tester.New(0, veryLargeGCInterval, veryLargeGCWindow, 1)
+	require.NoError(t, err)
+
+	ds, _ := testfixtures.StandardDatastoreWithData(rawDS, require.New(t))
+
+	// Run multiple registrations of the counter in parallel and ensure only
+	// one succeeds.
+	var numSucceeded atomic.Int32
+	var numFailed atomic.Int32
+	var wg sync.WaitGroup
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		go func() {
+			_, err = ds.ReadWriteTx(context.Background(), func(ctx context.Context, tx datastore.ReadWriteTransaction) error {
```

Comment:
there seems to be a data race due to all goroutines writing to the shared err variable.
Quick workaround: introducing a new variable inside the goroutine:
``` 
localErr := ds.ReadWriteTx(context.Background(), func(ctx context.Context, tx datastore.ReadWriteTransaction) error {

Reply:
Good catch. Fixed